inference-engine/src/inference_engine/ie_network_reader.cpp

```diff
@@ -198,14 +199,19 @@ CNNNetwork details::ReadNetwork(const std::string& modelPath, const std::string&
 #else
                 std::string weights_path = bPath;
 #endif
-                std::ifstream binStream;
-                binStream.open(weights_path, std::ios::binary);
-                if (!binStream.is_open())
+                std::shared_ptr<IAllocator> allocator = details::make_mmap_allocator(weights_path.c_str());
+                void* data = allocator->alloc(0);
+
+                if (data == nullptr)
                     THROW_IE_EXCEPTION << "Weights file " << bPath << " cannot be opened!";
 
+                details::MmapAllocator* mmap = reinterpret_cast<details::MmapAllocator*>(allocator.get());
+                const Blob::CPtr weights = make_shared_blob<uint8_t>({Precision::U8, { mmap->size() }, C }, reinterpret_cast<unsigned char*>(data));
+
                 // read model with weights
-                auto network = reader->read(modelStream, binStream, exts);
-                modelStream.close();
+                CNNNetwork network = reader->read(modelStream, weights, exts);
```

Comment:
@ilyachur @Drizshko can this method somehow hide weights the blob inside the network? E.g. via ProxyBlob like @ilyachur  suggested on the meeting. 

In general, I want to avoid all changes in public API like we agreed on the meeting. The feature look internal and API should not changed.

Reply:
But this one is not a public method? A meant it's not exposed to external APIs.

Reply:
Yes, I mean we use `setWeightsAllocator` later..